pilot/pkg/model/service.go

```diff
@@ -777,6 +791,10 @@ func (s *ServiceAttributes) Equals(other *ServiceAttributes) bool {
 		return false
 	}
 
+	if !slices.Equal(s.Aliases, other.Aliases) {
```

Comment:
does the order matter? Seems not

Reply:
We do the same order-dependant check for other slices. It does matter, since the order impacts XDS generation order. Although currently, no place we call Equals cares, it will be important if we did. They are NOT the same if the alias order is different, even if similar